docs/guides/debugging.md

```diff
@@ -0,0 +1,112 @@
+# Debug monitor
+
+To assist with creating and debugging complex builds, Buildx provides a
+debugger to help you step through the build process and easily inspect the
+state of the build environment at any point.
+
+> **Note**
+>
+> The debug monitor is a new experimental feature in recent versions of Buildx.
+> There are rough edges, known bugs, and missing features. Please try it out
+> and let us know what you think!
+
+## Starting the debugger
+
+To start the debugger, first, ensure that `BUILDX_EXPERIMENTAL=1` is set in
+your environment.
+
+```console
+$ export BUILDX_EXPERIMENTAL=1
+```
+
+To start a debug session for a build, you can use the `--invoke` flag with the
+build command to specify a command to launch in the resulting image.
+
+```console
+$ docker buildx build --invoke /bin/sh .
++] Building 4.2s (19/19) FINISHED
+ => [internal] connecting to local controller                                                                                   0.0s
+ => [internal] load build definition from Dockerfile                                                                            0.0s
+ => => transferring dockerfile: 32B                                                                                             0.0s
+ => [internal] load .dockerignore                                                                                               0.0s
+ => => transferring context: 34B                                                                                                0.0s
+ ...
+Launching interactive container. Press Ctrl-a-c to switch to monitor console
+Interactive container was restarted with process "dzz7pjb4pk1mj29xqrx0ac3oj". Press Ctrl-a-c to switch to the new container
+Switched IO
+/ #
+```
+
+This launches a `/bin/sh` process in the final stage of the image, and allows
+you to explore the contents of the image, without needing to export or load the
+image outside of the builder.
+
+For example, you can use `ls` to see the contents of the image:
+
+```console
+/ # ls
+bin    etc    lib    mnt    proc   run    srv    tmp    var
+dev    home   media  opt    root   sbin   sys    usr    work
+```
+
+## Using the monitor
+
+By default, when debugging, you'll be dropped into a shell in the final stage.
+
+By using the `Ctrl-a-c` key combination when in a debugging shell, you'll
+toggle between this debug shell and the monitor mode for the debugging session,
+which allows you to run high-level commands to control the debug environment.
```

Comment:
I'm guessing `a` is for `Alt`?

```suggestion
When you're in a debug shell, you can use the `Ctrl-Alt-C` key combination to
toggle between the debug shell and the monitor mode.
In monitor mode, you can run commands that control the debug environment.
```

Reply:
Ah it's actually not - it's CTRL+A, and then C on the keyboard. https://github.com/docker/buildx/pull/1168#pullrequestreview-1006893745.

I *think* this is inspired by QEMU. CTRL+A enters a "command" view, and then next character indicates what to do, e.g. `c` might enter a console, `x` might kill the process, etc.

Good callout, I'm not sure how best to phrase this clearly :cry:

Reply:
OK, got it. Sorry for the confusion. Not sure what the best way to put this is either. Vimmer in me wants to put something like:

the key combination `<C-a>c` (`Ctrl+A` followed by a `C`) opens the monitor mode

Reply:
> When you're in a debug shell, you can use the `Ctrl-a-c` key combination (press
`Ctrl`+`a` together, lift, then press `c`) to toggle between the debug shell
and the monitor mode. In monitor mode, you can run commands that control the
debug environment.

Hopefully that's descriptive enough :tada: